dali/imgcodec/parsers/pnm.cc

```diff
@@ -18,13 +18,60 @@ namespace dali {
 namespace imgcodec {
 
 ImageInfo PnmParser::GetInfo(ImageSource *encoded) const {
+  auto stream = encoded->Open();
+  ssize_t length = stream->Size();
+
+  // http://netpbm.sourceforge.net/doc/ppm.html
+  stream->Skip();
+  int channels = 1;
+  auto magic = stream->ReadOne<char>();
+  if ((magic == '3') || (magic == '6')) {
+    channels = 3;             // formats "p3" and "p6" are RGB color, all
+                              // other formats are bitmaps or greymaps
+  }
+
+  int state = 0;
+  enum {
+    STATE_START = 0, STATE_WIDTH = 1, STATE_HEIGHT = 2, STATE_DONE = 3
+  };
+  std::array<int, 2> dim = {0, 0};
+  auto cur = stream->ReadOne<char>();
+  do {
+    // comments can appear in the middle of tokens, and the newline at the
+    // end is part of the comment, not counted as whitespace
+    // http://netpbm.sourceforge.net/doc/pbm.html
+    if (cur == '#') {
+      do {
+        cur = stream->ReadOne<char>();
+      } while (cur != '\n');
+      cur = stream->ReadOne<char>();
+    } else if (isspace(cur)) {
+      if (++state < STATE_DONE) {
+        do {
+          cur = stream->ReadOne<char>();
+        } while (isspace(cur));
+      }
+    } else {
+      DALI_ENFORCE(isdigit(cur));
+      DALI_ENFORCE(state);
```

Comment:
I'd gladly see a proper exception instead of DALI_ENFORCE - and even with that, some error messages would be welcome.
```suggestion
      DALI_ENFORCE(isdigit(cur), "Expected a digit");
      DALI_ENFORCE(state, "Missing whitespace between PNM magic and image dimensions");
```

Reply:
While I agree that an exception makes sense here, does it change anything if exceptions aren't consistently used? You can't really expect to catch an error while using `GetInfo` as `TiffParser` already uses `DALI_ENFORCE`.